### PR TITLE
spring32

### PR DESCRIPTION
**Pull request by DemPaul**

### CODE REVIEW

File: src/main/java/onlineshop/spring/controller/InitController.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        this.userService = userService;
+        this.productService = productService;
+    }
+
+    @ModelAttribute("user")
+    public User setUpUserForm() {
+        return new User();
+    }
+
+    @RequestMapping(path = {"/"}, method = RequestMethod.GET)
```
Comment (SevostyanovGerman) [inline src/main/java/onlineshop/spring/controller/InitController.java]: init via PostConstruct

File: src/main/java/onlineshop/spring/controller/OrderController.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        } else {
+            model.addAttribute("missingBasketError",
+                    "Your basket does not exist, select the products and make a new one!");
+            return "forward:/user/product/all";
+        }
+    }
+
+    @RequestMapping(method = RequestMethod.POST)
+    public String orderPost(@SessionAttribute("user") User user,
+                            @RequestBody MultiValueMap<String, String> formData, Model model) {
```
Comment (SevostyanovGerman) [inline src/main/java/onlineshop/spring/controller/OrderController.java]: model

File: src/main/java/onlineshop/spring/controller/UserController.java [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    }
+
+    @RequestMapping(path = {"/add"}, method = RequestMethod.GET)
+    public String addUserGet() {
+        return "user-add";
+    }
+
+    @RequestMapping(path = {"/add"}, method = RequestMethod.POST)
+    public String addUserPost(@RequestBody MultiValueMap<String, String> formData, Model model) {
+        String email = formData.getFirst("email");
```
Comment (SevostyanovGerman) [inline src/main/java/onlineshop/spring/controller/UserController.java]: model

File: src/main/java/onlineshop/spring/controller/UserController.java [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+            return "user-edit";
+        } else {
+            return "redirect:/admin/user/all";
+        }
+    }
+
+    @RequestMapping(path = {"/edit"}, method = RequestMethod.POST)
+    public String editUserPost(@RequestParam("id") Long id,
+                               @RequestBody MultiValueMap<String, String> formData, Model model) {
+        String email = formData.getFirst("email");
```
Comment (SevostyanovGerman) [inline src/main/java/onlineshop/spring/controller/UserController.java]: del

File: src/main/java/onlineshop/spring/service/impl/MailServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+    private static final Logger mailServiceLogger = Logger.getLogger(MailServiceImpl.class);
+
+    @Override
+    public void sendConfirmCode(Code code) {
+
+        final String username = "temail884@gmail.com";
+        final String password = "12Qwert12";
+
+        Properties prop = new Properties();
```
Comment (Sofasmile) [inline src/main/java/onlineshop/spring/service/impl/MailServiceImpl.java]: naming

**Pull request merged**